syntax = "proto3";
package zht_im;
option cc_generic_services = true;
message SpeechRecognitionReq {
    string request_id = 1;                  // 请求ID
    bytes speech_content = 2;               // 语音数据
    optional string user_id = 3;            // 用户ID
    optional string session_id = 4;         // 登录会话ID -- 用于网关进行身份鉴权
}
message SpeechRecognitionRsp {
    string request_id = 1;                  // 请求ID
    bool success = 2;                       // 处理结果标志
    optional string errmsg = 3;             // 失败原因
    optional string recognition_result = 4; // 识别后的文字数据
}

// 语音识别Rpc服务及接口的定义
service SpeechService {
    rpc SpeechRecognition(SpeechRecognitionReq) returns (SpeechRecognitionRsp);
}